examples/Parser.hs

```diff
@@ -100,9 +101,9 @@ parse input = (>>= exhaustive) . runState input . runParseC
         exhaustive _       = empty
 
 newtype ParseC m a = ParseC { runParseC :: StateC String m a }
-  deriving newtype (Alternative, Applicative, Functor, Monad)
+  deriving (Alternative, Applicative, Functor, Monad)
 
-instance (Alternative m, Algebra sig m) => Algebra (Symbol :+: sig) (ParseC m) where
+instance (Alternative m, Algebra sig m, CanHandle sig ((,) String)) => Algebra (Symbol :+: sig) (ParseC m) where
```

Comment:
This sort of thing is common enough for algebras that I’ve been considering defining a constraint synonym along these lines:

```haskell
type AlgebraFor ctx sig m = (Algebra sig m, CanHandle sig ctx)
```

which would in turn tidy up the instance contexts a bunch:

```haskell
instance (Alternative m, AlgebraFor ((,) String) sig m) => Algebra (Symbol :+: sig) (ParseC m) where
```

I don’t think `AlgebraFor` is a very good name, however. `AlgebraHandling`? (`AlgebraThreading` if we accept #300?) I don’t know.